src/mame/drivers/silverball.cpp

```diff
@@ -0,0 +1,130 @@
+// license:BSD-3-Clause
+// copyright-holders:
+/*******************************************************************************
+
+    Skeleton driver for TAB Austria "Silverball" PC-based touch games
+
+
+    Hardware overview for "Silverball (8.01)":
+    Motherboard: Soyo 5EAS / Soyo SY-7IZB+ / Acorp 694XT-694XT1 / others
+    32768K RAM (4 x 8MB)
+    Intel Pentium-S CPU 133 MHz (A80502133)
+
+    MicroTouch (external)
+    ESS AudioDrive ES1869FC ISA sound card
+    S3 Trio64V2/DX PCI VGA (86C775, 512KB RAM)
+    I/O Controller Silverball 3 (ISA card, Actel A40MX04-F PL84, DB50 external connector)
+
+    HDD Samsung Maxtor G1021U2 or other IDE HDD with similar capacity.
+
+    The manual points to a HardLock parallel port security dongle, but it was missing
+    from the dumped machine.
+
+*******************************************************************************/
+
+#include "emu.h"
+#include "cpu/i386/i386.h"
+#include "speaker.h"
+
+namespace {
+
+class silverball_state : public driver_device
+{
+public:
+	silverball_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag)
+		, m_maincpu(*this, "maincpu")
+	{
+	}
+
+	void silverball(machine_config &config);
+
+private:
+	void mem_map(address_map &map);
+	void io_map(address_map &map);
+
+	required_device<cpu_device> m_maincpu;
+};
+
+void silverball_state::mem_map(address_map &map)
+{
```

Comment:
Can you please add at least conventional memory and base bios regions, i.e.:

```cplusplus
	map(0x00000000, 0x0009ffff).ram();
	map(0xfffc0000, 0xffffffff).rom().region("mb_bios", 0);
```
I don't want to compile just for this simple step, thanks in advance.

Reply:
If I add those two lines I got:
Intel Pentium device 'maincpu': program space is 32-bit but references 8-bit region ':mb_bios'

Can you please help? Thx!

Reply:
ROM_REGION needs to be ROM_REGION32_LE

Reply:
Works fine now, thanks!

Reply:
> 
> 
> Can you please add at least conventional memory and base bios regions here, i.e.:
> 
> ```
> 	map(0x00000000, 0x0009ffff).ram();
> 	map(0xfffc0000, 0xffffffff).rom().region("mb_bios", 0);
> ```
> 
> I don't want to compile just for this simple step, thanks in advance.

@angelosa Can you please explain it? Just to learn for future PC skeleton drivers.

For the BIOS, from 0xfffc0000 to 0xffffffff it's 256K, the bigger BIOS size, it's after that? Why on the upper size? Does the BIOS always maps on the upper area? Why on 0xffffffff?

For the RAM, 640K... I'd say it's the original PC RAM limit that maps at the start, is it?

Thanks!!